GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -824,15 +824,22 @@ private void SaveSelectedItemToTempFile(Action<string> onSaved)
                 await TaskScheduler.Default;
 
                 var blob = Module.GetFileBlobHash(item.Item.Name, item.SecondRevision.ObjectId);
-
                 if (blob is null)
                 {
                     return;
                 }
 
                 var fileName = PathUtil.GetFileName(item.Item.Name);
                 fileName = (Path.GetTempPath() + fileName).ToNativePath();
-                Module.SaveBlobAs(fileName, blob.ToString());
+                try
+                {
+                    Module.SaveBlobAs(fileName, blob.ToString());
+                }
+                catch (Exception ex)
+                {
+                    await this.SwitchToMainThreadAsync();
+                    ThrowUserExternalOperationException(ex.Message, command: null, arguments: item.Item.Name, ex);
```

Comment:
This shouldn't be necessary. Rather, you can update `FileAndForget` to catch this exception and report it on a UI thread (this is the purpose of `FileAndForget` as opposed to a plain `Forget` method).

Reply:
Thank you for pointing this out, @sharwell! It works right out of the box because `FileAndForget` already catches exceptions, switches to the UI thread and calls `Application.OnThreadException`.

@RussKie, in which context must an exception been thrown so that it is reported via `AppDomain.CurrentDomain.UnhandledException` only?

Reply:
If an exception is thrown from a non-UI thread it will get handled by `AppDomain.CurrentDomain.UnhandledException` handler. 
Docs: https://docs.microsoft.com/dotnet/api/system.windows.forms.unhandledexceptionmode